crates/transaction-pool/src/pool/pending.rs

```diff
@@ -279,6 +280,87 @@ impl<T: TransactionOrdering> PendingPool<T> {
         self.remove_transaction(&worst)
     }
 
+    /// Truncates the pool to the given limit.
+    pub(crate) fn truncate_pool(
```

Comment:
all of this looks more or less identical to the other truncate impl. 

I wonder if the approach is to apply this to the `AllTransactions` set so we no longer need to enforce this on a per subpool basis?

Reply:
Yeah the current impl logic is same for both the sub-pools. I'm working on to mirror the algorithm as used in geth (which are different for `Pending` and `Parked` subpool)